File: test/2020winter/p1.java
class Solution {
    public String solution(int n, int[][] delivery) {
        String answer = "";
        
        int [] check = new int[n + 1];
        
        for(int i = 0; i < delivery.length; i++) {
            if(delivery[i][2] == 1) {
                check[delivery[i][0]] = 1;
                check[delivery[i][1]] = 1;
            }
        }
        
        for(int i = 0; i < delivery.length; i++) {
            if(delivery[i][2] == 0) {
                if(check[delivery[i][0]] == 1) {
                    check[delivery[i][1]] = -1;
                } else if(check[delivery[i][1]] == 1) {
                    check[delivery[i][0]] = -1;
                }
            }
        }
        
        for(int i = 1; i <= n; i++) {
            if(check[i] == 1) answer += "O";
            else if(check[i] == 0) answer += "?";
            else answer += "X";
        }
        
        return answer;
    }
}

File: test/2021N/p1.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

string solution(int n, vector<int> p, vector<int> c) {
    string answer = "";
    int failDay = 0;
    int cost = 100;
    double totalCost = 0;
    double day = 0;

    for(int i = 0; i < n; i++) {

        if(failDay == 3) break;

        if(p[i] >= c[i]) {
            totalCost += (c[i] * cost);

            if(i+1 < n) 
                p[i+1] += (p[i]-c[i]);

            failDay = 0;
            cost = 100;
        } else {
            failDay++;

            if(i+1 < n)
                p[i+1] += p[i];

            cost/=2;
        }

        day++;
    }

    double average = totalCost / day;

    average = (round(average * 100)) / 100.0;

    answer = to_string(average);
    answer = answer.substr(0, answer.length()-4);

    return answer;
}